tests/system/feature_store/test_spark_engine.py

```diff
@@ -1524,6 +1524,75 @@ def test_ingest_and_get_offline_features_with_hdfs(self):
 
         resp_df.reset_index(drop=True, inplace=True)
         pd.testing.assert_frame_equal(resp_df[["bad", "department"]], expected_df)
+        target.purge()
+        with pytest.raises(FileNotFoundError):
+            target.as_df()
+        target.purge()
+
+    @pytest.mark.skipif(
+        not {"HDFS_HOST", "HDFS_PORT", "HDFS_HTTP_PORT", "HADOOP_USER_NAME"}.issubset(
+            os.environ.keys()
+        ),
+        reason="HDFS host, ports and user name are not defined",
+    )
+    def test_hdfs_wrong_credentials(self):
+        host = os.environ.pop("HDFS_HOST")
+        try:
+            datastore_profile = DatastoreProfileHdfs(
+                name="my-hdfs",
+                host="localhost",
+                port=int(os.getenv("HDFS_PORT")),
+                http_port=int(os.getenv("HDFS_HTTP_PORT")),
+            )
+            register_temporary_client_datastore_profile(datastore_profile)
+            self.project.register_datastore_profile(datastore_profile)
+            target = ParquetTarget(
+                "mytarget", path=f"{self.hdfs_output_dir}-get_offline_features"
+            )
+            with pytest.raises(ConnectionError):
+                target.purge()
+        finally:
+            os.environ["HDFS_HOST"] = host
+
+    @pytest.mark.skipif(
+        not {"HDFS_HOST", "HDFS_PORT", "HDFS_HTTP_PORT", "HADOOP_USER_NAME"}.issubset(
+            os.environ.keys()
+        ),
+        reason="HDFS host, ports and user name are not defined",
+    )
+    def test_hdfs_empty_credentials(self):
+        host = os.environ.pop("HDFS_HOST")
+        port = int(os.environ.pop("HDFS_PORT"))
+        http_port = int(os.environ.pop("HDFS_HTTP_PORT"))
+        try:
+            datastore_profile = DatastoreProfileHdfs(
+                name="my-hdfs",
+            )
+            register_temporary_client_datastore_profile(datastore_profile)
+            self.project.register_datastore_profile(datastore_profile)
+            target = ParquetTarget(
+                "mytarget", path=f"{self.hdfs_output_dir}-get_offline_features"
+            )
+            with pytest.raises(ValueError):
+                target.purge()
+
+            datastore_profile = DatastoreProfileHdfs(
+                name="my-hdfs",
+                port=port,
+                http_port=http_port,
+            )
+            register_temporary_client_datastore_profile(datastore_profile)
+            self.project.register_datastore_profile(datastore_profile)
+            target = ParquetTarget(
+                "mytarget", path=f"{self.hdfs_output_dir}-get_offline_features"
+            )
+            with pytest.raises(ConnectionError):
```

Comment:
Here too, it looks like the problem is wrong or missing host and not the credentials.

Reply:
if you do not provide any user it just use some default, so I choose to remove host.  do you have another idea?

Reply:
Yup, to name the test accordingly :)